backends/ceed-backend-weak.c

```diff
@@ -2,16 +2,31 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-// This function provides a debug target for weak symbols
 // LCOV_EXCL_START
-static int CeedRegister_Weak(const char *name) {
+// This function provides improved error messages for uncompiled backends
+static int CeedInit_Weak(const char *resource, Ceed ceed) {
+  return CeedError(ceed, CEED_ERROR_UNSUPPORTED,
+                   "Backend not currently compiled: %s", resource);
+}
```

Comment:
That's great, how difficult would it be to let backends personalize the error message? But I think this is great already.

Reply:
I mean, technically we could hack it in as extra information in the string holding the resource name in the MACRO. Not sure if I like it, but it'll *work*.

Reply:
It could take pairs of strings, one stating the resource name and the other explaining which make options are needed (`make CUDA_DIR=/path/to/cuda`), but I think this is hard to test and wrong/outdated information is worse than less detailed information that accurately identifies the problem.

Reply:
We could just change the message to
```
$ ./build/t000-ceed /cpu/self/xsmm/serial
/home/jeremy/Dev/libCEED/backends/ceed-backend-weak.c:9 in CeedInit_Weak(): Backend not currently compiled: /cpu/self/xsmm/serial
Consult the installation instructions to compile this backend
Aborted (core dumped)
```

Reply:
done